internal/service/networkpeering/networkpeering.go

```diff
@@ -296,14 +296,20 @@ func waitForPeeringToLeaveProvisionedState(ctx context.Context, svc *service.Ser
 	for {
 		select {
 		case <-ctx.Done():
-			diags.AddError("Context cancelled", ctx.Err().Error())
+			diags.AddError(
+				"Context cancelled",
+				utils.ErrorDiagnosticDetail(ctx.Err()),
+			)
 			return
 		default:
 			peering, err := svc.GetNetworkPeering(ctx, &request.GetNetworkPeeringRequest{
 				UUID: uuid,
 			})
 			if err != nil {
-				diags.AddError("Unable to get network peering details", ctx.Err().Error())
+				diags.AddError(
```

Comment:
Not related, but caused segfault in the acceptance tests triggered by this PR, so might as well fix here 🙈